Update for react-reconciler 0.26.1, add "color" prop

unframework opened the pull request:
Hey, thanks for setting up this library!

I was trying to use it on a recent react-reconciler version and got an infinite loop crash - this PR adds a couple of missing callbacks that seem to be required in newer versions.

I also added a "color" prop that runs `setColor` on shapes. Finally, there was a bit of tweaking done to the root container instance, to allow convenience union logic.

By the way, I have a custom fork in my local repo that uses a "geometry" helper instead of the Model/Mesh helpers, so that it can be plugged into any arbitrary parent mesh with materials (but does not expose the intermediate "parts", just final polygons) - happy to share that too as it might add some flexibility.

szymonkaliski commented on packages/modeler-csg/src/reconciler.js:
```diff
@@ -113,23 +135,36 @@ const CSGRenderer = new Reconciler({
   appendChild(parent, child) {
-    parent.appendChild(child);
+    // no subsequent mutations allowed
+    throw new Error('not supported');
```
should we maybe have bit more info in the error?

szymonkaliski commented on packages/modeler-csg/src/reconciler.js:
```diff
@@ -113,23 +135,36 @@ const CSGRenderer = new Reconciler({
   removeChild(parent, child) {
-    console.warn("removeChild not implemented!", { parent, child });
+    // no subsequent mutations allowed
+    throw new Error('not supported');
```
should we maybe have bit more info in the error?

szymonkaliski commented:
Hey @unframework - thanks so much - awesome PR!
I left some comments about errors, maybe we should try to leave bit more info when things fail.

Re: Color - are there any other props worth supporting? Tbh. I only use CSG for generating the mesh, and then apply materials with color elsewhere.

Out of curiosity - what are you using this for?

> By the way, I have a custom fork in my local repo that uses a "geometry" helper instead of the Model/Mesh helpers, so that it can be plugged into any arbitrary parent mesh with materials (but does not expose the intermediate "parts", just final polygons) - happy to share that too as it might add some flexibility.

Yea please share, I'd love to have a look - we can figure out if it makes sense to merge with this repo or not.

unframework commented:
Thanks for the prompt response Szymon!

I tried to expand error message wording, hopefully it is a bit better now. I also fixed code style (quotes/etc) to match the rest of the file. In addition, I added some README copy to mention the `color` property and also to mention `jscad/csg.js` a bit more prominently (I hope that's ok - it was actually very helpful for me as a newbie to understand what the underlying math engine was).

I did not see anything else beside `color` to add yet, but maybe that will change in the future. Either way, I will try to keep any further feature suggestions in a separate PR if that's OK? And re: my use cases, I am experimenting with live-generating geometry for a [lightmap pass with my baker](https://github.com/unframework/threejs-lightmap-baker) and some kind of mini-FPS demo walkthrough.

Cheers!

szymonkaliski commented:
Awesome, thanks so much - merging this in!
